transports/quic/src/provider/async_std.rs

```diff
@@ -59,4 +60,16 @@ impl super::Provider for Provider {
     fn sleep(duration: Duration) -> BoxFuture<'static, ()> {
         async_std::task::sleep(duration).boxed()
     }
+
+    fn from_std_udp_socket(socket: std::net::UdpSocket) -> io::Result<Self::UdpSocket> {
+        Ok(socket.into())
+    }
+
+    fn send_to<'a>(
+        udp_socket: &'a Self::UdpSocket,
+        buf: &'a [u8],
+        target: std::net::SocketAddr,
+    ) -> BoxFuture<'a, io::Result<usize>> {
+        udp_socket.send_to(buf, target).boxed()
+    }
```

Comment:
Instead of two functions, we could also make one function that accepts an `std::net::UdpSocket`, converts it internally to the runtime-specific one and sends, the packet and returns the socket.

That would avoid the associated type and reduce this to one function that needs to be implemented.